### ISSUE TITLE
Get motile on conda-forge

### ISSUE THREAD

**Issue by cmalinmayor** [funkelab/motile#98]
@tlambert03 @constantinpape mentioned that the Gurobi dependency is not necessary for his work and it is limiting to have to install it with the extra conda channels. I know we discussed looking into not needing to install Gurobi - can you remind me how hard this would be to implement logistically?

**Comment by tlambert03** [comment]
It needs a change at the c level in ilpy.  It's conceptually straightforward, but not trivial.

https://github.com/funkelab/ilpy/issues/14

**Comment by tlambert03** [comment]
Hi @constantinpape,
Apologies, it's not on conda forge... it still represents a significant amount of work that I don't currently have time for.  I'll reopen this.

**Comment by constantinpape** [comment]
Thanks for letting me know @tlambert03 and thanks for reopening the issue.

I understand that this is quite some work, just wanted to express that there is still interest from our side in this.
(Without a conda-forge installation it is quite cumbersome to integrate this as dependency with other conda packages, since conda-forge builds cannot refer to external channels.)

**Comment by tlambert03** [comment]
Yeah I know, sorry.  It's precisely because they can't refer to other channels that makes this hard, because the c code ilpy was originally written in such a way that requires the gurobi library to be present (even if you don't intend to use it).  So, putting this on conda forge requires rewriting ilpy to dynamically load libraries rather than link them upfront.  It's doable, but takes some surgery on ilpy

**Comment by constantinpape** [comment]
I discussed this with @funkey last week, and he said that the necessary prerequisites are now done, and that motile can now be installed without gurobi channel.
If that's the case, could we now get `ilpy` and `motile` on conda-forge? Then we could fully integrate it in our dependencies :).

**Comment by tlambert03** [comment]
are you referring to this? https://github.com/funkelab/ilpy/pull/70

I'd say that we have a proof of principle, but not that the pre-requisites are done.  It would be great if we could get some people trying out that PR.  I have concerns that the scip version specifically is slower than the current code (but exactly how much is something I haven't been able to check).  Anyway, that PR is the path towards conda-forge... but it's still in need of some feedback and probably tuning

**Comment by tlambert03** [comment]
if you have a moment to check out that branch and try it on your stuff, your feedback would be appreciated!

you might be able to just `pip install git+https://github.com/tlambert03/ilpy@pure-py`

**Comment by constantinpape** [comment]
I probably won't have time, but we have two students in the lab who will start to work on tracking, I will ask them to give this a try once they are up-to-speed.

**Comment by tlambert03** [comment]
thanks.  Yeah, basically since that is more or less a complete rewrite, it has lots of implications and I don't want to just merge and publish it without someone trying it on some real data.  @cmalinmayor, would be good to get your input on that as well

**Comment by cmalinmayor** [comment]
> thanks. Yeah, basically since that is more or less a complete rewrite, it has lots of implications and I don't want to just merge and publish it without someone trying it on some real data. [@cmalinmayor](https://github.com/cmalinmayor), would be good to get your input on that as well

Yes, I'd like to test it before depending on the pure python version, but I don't currently have the bandwidth to do full integration testing. I will try next week or the week after to implement a benchmarking script that can be run via github action so that we can monitor performance impact of this and future changes.

**Comment by constantinpape** [comment]
This is amazing @tlambert03 ! Will test that it works soon.
